api/v1alpha1/bucket_types.go

```diff
@@ -0,0 +1,38 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+type Bucket struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BucketSpec   `json:"spec,omitempty"`
+	Status BucketStatus `json:"status,omitempty"`
+}
+
+type BucketSpec struct {
+	Name               string                   `json:"name"`
+	CreationSecret     corev1.SecretKeySelector `json:"bucketCreationSecret"`
```

Comment:
NIT: `bucketCreationSecret` is overdescriptive on the JSON tag. Just `creationSecret`